dashboard/app/controllers/api/v1/pd/workshops_controller.rb

```diff
@@ -108,6 +108,38 @@ def upcoming_teachercons
     render json: {error: e.message}, status: :bad_request
   end
 
+  def to_geojson(workshops)
```

Comment:
Out of curiosity, why is this new function needed?

Reply:
Mapbox works well with geojson (a json standard for locations) so I just pushed all the work off to that library when creating the sources/layers.